### PR TITLE
[REF] web: remove LegacyComponent

### PR DESCRIPTION
**Pull request by mcm-odoo**
Before this commit, components used `el` and `trigger` but this works
thanks to LegacyComponent. LegacyComponent has been introduced to ease
the transition from owl 1 to owl 2.

This commit replaces the uses of `el` by `useRef` and `trigger` by
adding callback in props. Without these owl 1 features, components
no longer need to extend LegacyComponent.
This commit also adapts the tests to use `getFixture` instead of `el`.

Co-authored-by: Aaron Bohy <aab@odoo.com>
Co-authored-by: Simon Genin (ges) <ges@odoo.com>
Co-authored-by: luvi <luvi@odoo.com>
Co-authored-by: Jorge Pinna Puissant <jpp@odoo.com>

### CODE REVIEW

File: addons/web/static/src/webclient/navbar/navbar.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ export class NavBar extends LegacyComponent {
      *     By the end of this method another render may occur depending on the adaptation result.
      */
     async adapt() {
-        if (!this.el) {
+        if (!this.rootRef.el) {
```
Comment (sdegueldre) [inline addons/web/static/src/webclient/navbar/navbar.js:98]: comment to check that we still need this

File: addons/web/static/tests/webclient/navbar_tests.js [hunk 2]
```diff
@@ -241,7 +235,7 @@ QUnit.test("can adapt with 'more' menu sections behavior", async (assert) => {
     ]);
 });
 
-QUnit.test(
+QUnit.skip(
```
Comment (sdegueldre) [inline addons/web/static/tests/webclient/navbar_tests.js:238]: skipped test

**Review by sdegueldre — CHANGES_REQUESTED** [review]

File: addons/web/static/src/views/graph/graph_renderer.js [hunk 3]
```diff
@@ -39,10 +38,11 @@ function shortenLabel(label) {
     return shortLabel;
 }
 
-export class GraphRenderer extends LegacyComponent {
+export class GraphRenderer extends Component {
     setup() {
         this.model = this.props.model;
 
+        this.rootRef = useRef("root");
```
[the remark this answers is not in the document]
Comment (mcm-odoo) [reply addons/web/static/src/views/graph/graph_renderer.js:45]: I didn't change this one to keep consistency with other refs on lines just below

**Comment by sdegueldre** [comment]
@robodoo r+

**Pull request closed without merging**